Fix build using musl libc

crater2150 opened the pull request:
While packaging the current version for Void Linux (https://github.com/void-linux/void-packages/pull/31123) I noticed the build failing when compiling against [musl](https://musl.libc.org/). This is fixed by including `fcntl.h` for the definition of `open`.

JoseExposito commented on src/utils/client-lock.cpp:
```diff
@@ -19,6 +19,7 @@
 
 #include <sys/file.h>
 #include <unistd.h>
+#include <fcntl.h>
```
The linter can be a little bit nitpicky:

```diff
+ #include <fcntl.h>
#include <sys/file.h>
#include <unistd.h>
- #include <fcntl.h>
```

JoseExposito reviewed (changes requested):
Hi!

Thanks a lot for sending the patch upstream.

Do you mind to change the include order, please? I try to keep them sort alphabetically and make the linter happy.

JoseExposito commented:
Oh! And feel free to add install instructions for VoidLinux in the README.md file. If you are busy, just tell me the command and I'll add it :D

crater2150 commented:
Thanks. I changed the include order and added a paragraph to the readme. But note that the part about enabling the service will only be valid as soon as https://github.com/void-linux/void-packages/pull/31123 is merged, as the current package doesn't yet include a service file for Void's init system.

JoseExposito reviewed (approved):
Hi,

Thanks for documenting the installation, and for submitting the patch upstream.

This kind of bugs are very difficult to catch (too many distros to test), so I truly appreciate that you took the time to upstream the fix.

Best wishes,
Jose
